kanidmd/score/src/https/oauth2.rs

```diff
@@ -219,12 +226,43 @@ async fn oauth2_authorise(
         .await;
 
     match res {
-        Ok(consent_req) => {
+        Ok(AuthoriseResponse::ConsentRequested {
+            client_name,
+            scopes,
+            pii_scopes,
+            consent_token,
+        }) => {
             // Render a redirect to the consent page for the user to interact with
             // to authorise this session-id
             let mut res = tide::Response::new(200);
             // This is json so later we can expand it with better detail.
-            tide::Body::from_json(&consent_req).map(|b| {
+            tide::Body::from_json(&AuthorisationResponse::ConsentRequested {
+                client_name,
+                scopes,
+                pii_scopes,
+                consent_token,
+            })
+            .map(|b| {
+                res.set_body(b);
+                res
+            })
+        }
+        Ok(AuthoriseResponse::Permitted(AuthorisePermitSuccess {
+            mut redirect_uri,
+            state,
+            code,
+        })) => {
+            let mut res = tide::Response::new(302);
+
+            redirect_uri
+                .query_pairs_mut()
+                .clear()
+                .append_pair("state", &state)
+                .append_pair("code", &code);
+            res.insert_header("Location", redirect_uri.as_str());
+            // I think the client server needs this
+            // res.insert_header("Access-Control-Allow-Origin", redirect_uri.origin().ascii_serialization());
```

Comment:
not a bad idea... tl;dr "don't share the response with any other servers"

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin

Reply:
Yeah. I need to see how it interacts with various bits, I'm not sure really.